pyxem/utils/indexation_utils.py

```diff
@@ -603,16 +555,42 @@ def _match_polar_to_library_cpu(
     N is the number of templates and R the number of spots in the template
     with the maximum number of spots
     """
-    correlations = _correlate_polar_to_library_cpu(
-        polar_image, r_templates, theta_templates, intensities_templates
-    )
-    correlations_mirror = _correlate_polar_to_library_cpu(
-        polar_image,
-        r_templates,
-        (polar_image.shape[0] - theta_templates) % polar_image.shape[0],
-        intensities_templates,
+    N = r_templates.shape[0]
+    D = r_templates.shape[1]
+    n_shifts = polar_image.shape[0]
+    best_in_plane_shift = np.empty(N, dtype=np.int32)
+    best_in_plane_shift_m = np.empty(N, dtype=np.int32)
+    best_in_plane_corr = np.empty(N, dtype=polar_image.dtype)
+    best_in_plane_corr_m = np.empty(N, dtype=polar_image.dtype)
+
+    for template in prange(N):
+        inplane_cor = np.zeros(n_shifts)
+        inplane_cor_m = np.zeros(n_shifts)
+        for spot in range(D):
+            rsp = r_templates[template, spot]
+            if rsp == 0:
+                break
+            tsp = theta_templates[template, spot]
+            isp = intensities_templates[template, spot]
+            spl = n_shifts - tsp
+            inplane_cor[:spl] += polar_image[tsp:, rsp] * isp
+            inplane_cor[spl:] += polar_image[:tsp, rsp] * isp
+            inplane_cor_m[:tsp] += polar_image[spl:, rsp] * isp
+            inplane_cor_m[tsp:] += polar_image[:spl, rsp] * isp
+
+        best_shift = np.argmax(inplane_cor)
+        best_shift_m = np.argmax(inplane_cor_m)
+        best_in_plane_shift[template] = best_shift
+        best_in_plane_shift_m[template] = best_shift_m
+        best_in_plane_corr[template] = inplane_cor[best_shift]
+        best_in_plane_corr_m[template] = inplane_cor_m[best_shift_m]
+
+    return (
```

Comment:
Since `_get_best_correlations_and_angles()` isn't used here anymore, it's only used in the GPU equivalent function. Would it be better in terms of maintenance and readability to just remove the function and use `cp.argmax()` etc. in the GPU function, as is done here with `np.argmax()`?

Reply:
Agreed. Done.